service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java

```diff
@@ -1056,30 +1064,22 @@ public JobProperties getJob(String jobId) throws IllegalArgumentException, IOExc
      */
     public Module addModule(Module module) throws IOException, IotHubException, JsonSyntaxException
     {
-        // Codes_SRS_SERVICE_SDK_JAVA_REGISTRYMANAGER_28_001: [The constructor shall throw IllegalArgumentException if the input module is null]
         if (module == null)
         {
             throw new IllegalArgumentException("module cannot be null");
         }
 
-        // Codes_SRS_SERVICE_SDK_JAVA_REGISTRYMANAGER_28_002: [The function shall deserialize the given module object to Json string]
         String moduleJson = module.toDeviceParser().toJson();
 
-        // Codes_SRS_SERVICE_SDK_JAVA_REGISTRYMANAGER_28_003: [The function shall get the URL for the module]
-        URL url = iotHubConnectionString.getUrlModule(module.getDeviceId(), module.getId());
-        // Codes_SRS_SERVICE_SDK_JAVA_REGISTRYMANAGER_28_004: [The function shall create a new SAS token for the module]
-        String sasTokenString = new IotHubServiceSasToken(this.iotHubConnectionString).toString();
+        URL url = IotHubConnectionString.getUrlModule(this.hostName, module.getDeviceId(), module.getId());
```

Comment:
This looks like repeated code pattern. Can we extract to a method like CreatHttpRequest()?

Reply:
Scratch that. It is not as repeated as I thought. Different Http requests and urls.

Reply:
However, it could all be done in a single line. But leave it to your preference.

Reply:
The particular URL to use changes per method, but I can factor out the next line since every request should have an authentication token